hybridse/src/udf/dynamic_lib_manager.cc

```diff
@@ -102,20 +92,18 @@ base::Status DynamicLibManager::ExtractFunction(const std::string& name, bool is
 }
 
 base::Status DynamicLibManager::RemoveHandler(const std::string& file) {
-    std::shared_ptr<DynamicLibHandle> so_handle;
+    void* handle = nullptr;
     {
         std::lock_guard<std::mutex> lock(mu_);
         if (auto iter = handle_map_.find(file); iter != handle_map_.end()) {
-            iter->second->ref_cnt--;
-            if (iter->second->ref_cnt == 0) {
-                so_handle = iter->second;
-                handle_map_.erase(iter);
+            if (iter->second != nullptr) {
```

Comment:
erase first

Reply:
a dynamic may be used by multi functions.